File: solutions/java/collatz-conjecture/1/src/main/java/CollatzCalculator.java
import java.util.stream.Stream;
class CollatzCalculator {

    int computeStepCount(int start) {
        if (start < 1) {
            throw new IllegalArgumentException("Only positive integers are allowed");
        }
        return (int) Stream.iterate(start, i -> i != 1, i -> i % 2 == 0 ? i / 2 : i * 3 + 1).count();
    }
}

File: solutions/kotlin/collatz-conjecture/1/src/main/kotlin/CollatzCalculator.kt
object CollatzCalculator {
    fun computeStepCount(start: Int): Int = start
        .also { require (start >= 1) { "Invalid input: Must provide positive integer" } }
        .let { generateSequence(start) { num ->
            if (0 == num and 1) {
                num shr 1
            } else {
                num + (num shl 1) + 1
            }
        }.takeWhile { it != 1 }.count() }
}